### ISSUE TITLE
Refactor settings: Allow direct input for sliders

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#12176]
### Describe your problem!

In our currently released versions you can enter a specific value for sliders when you tap on the number right beside the slider. Our refactored version does no longer support this. Don't know if there is a way to support this again, but would be nice if so.

### How to reproduce?

- open settings => map
- see sliders for proximity notification (and others)
- tap on the number beside one of the sliders

### Actual result after these steps?

nothing happens

### Expected result after these steps?

a type-specific input field should open, letting the user enter a value

### Reproducible

Yes

### c:geo Version

dev-build from current state of #11298

**Comment by fm-sys** [comment]
AFAIK this is no longer possible. At least if we don't want to build our own slider preference from scratch...

Instead of direct input, I guess we should rather use meaningful step values. E.g. the listloadlimit currently supports values from 0-50000. Why should a user ever select 7259 or any other random number? For that preference, 100th steps would be helpful.

The same idea applies to some more slider preferences.

**Comment by eddiemuc** [comment]
We have a custom slider view component in c:geo which I created back when I did the filters (to add the value display below it e.g. 1,2,3,4,5 as possible D/T values. This view could get extended easily by an optional field to click to enter a value directly.
Injust don't know how to do this UI wise. Maybe just place an icon button with "edit" Symbol (the pen) to the right of the slider?

**Comment by fm-sys** [comment]
> We have a custom slider view component in c:geo which I created back when I did the filters

The slider itself is not the problem. The difference between other places is that the preferences aren't using normal layout files AFAIK

**Comment by SchoolGuy** [comment]
@fm-sys We even had that custom slider in the old preferences already there and it would be possible with the new preferences. The problem is in my eyes that we allow things which are so custom but require an enormous amount of code. Imagine for example the abstraction with the Dialog: This was ~500 lines of code through 4 levels of abstract classes in the end to show just a dialog where the user then click "ok" or "more". And we used it 2 times in the preferences. Now I duplicated that 10 lines dialog code to the two places and we maintain 20 lines of code with no abstraction instead of ~500 with four levels of abstraction. Abstraction is good and I love it and I know you guys put a lot of thought in it but it is only useful if it saves us effort not generates it.

TLDR: I side with you in the way that we do want to set reasonable steps for the users.

**Comment by moving-bits** [comment]
While migrating the settings all of our `SeekbarPreference` items got reset to the default `SeekbarPreference`. This has a couple of downsides under certain conditions:
- Some sliders need to cover a wide range. For the end-user, the relevant values are probably non-linear - it may be relevant for you to distinguish between 15 and 20, but not between 975 and 1000. Thus the old component supported an exponential distribution of values (e. g.: 1-10 in the first third, 10-100 in the second, 100-1000 in the last third).
- If a slider is used for distances, it needs to take both metric and imperial units into account UI-wise. The old component used meters internally (or km, depending on config), and transformed it appropriately for the UI.
- If you want to have a specific value, it can be very hard to hit it exactly on a small touch space. That's why the old component supported a value entry field (which BTW was explicitly requested back at the time).

I'm fine with using the custom slider component that got created when implementing the filters, and having clickable step values may help, but I would really like to see the points listed above be taken into consideration as well.

**Comment by fm-sys** [comment]
> I'm fine with using the custom slider component that got created when implementing the filters

That would mean quite some extra work, as our current filter slider is not compatible with the preferences architecture at all.

But there is a way easier solution:

- Extend the androidx `SeekbarPreference` class
- Query the textView ID in the method where the view gets initialized (`androidX.R.id.*whatever*`) and add an onClickListener 

That saves us quite some work, follows material design and is easy to maintain.

For logarithmic scale, there is a pending feature request:
https://github.com/material-components/material-components-android/issues/1750

**Comment by moving-bits** [comment]
Ok, then I'll dive into `androidx.preference.SeekBarPreference` and try to accomplish what we had in earlier versions (where I had extended `android.preference.SeekBarPreference` to achieve the desired behavior).
